src/psycopt2d/train_model.py

```diff
@@ -0,0 +1,157 @@
+"""Training script for training a single model for predicting t2d.
+
+TODO:
+add split for using pre-defined train-val split
+multiple CV
+
+Features:
+# fix impute
+# move filter to compute
+"""
+
+from pathlib import Path
+from typing import Iterable, Optional
+
+import hydra
+import numpy as np
+import wandb
+from sklearn.metrics import roc_auc_score
+from sklearn.model_selection import StratifiedGroupKFold
+from sklearn.pipeline import Pipeline
+from wandb.sdk import wandb_run
+
+from psycopt2d.feature_transformers import ConvertToBoolean, DateTimeConverter
+from psycopt2d.load import load_dataset
+from psycopt2d.models import MODELS
+from psycopt2d.utils import flatten_nested_dict
+
+CONFIG_PATH = Path(__file__).parent / "config"
+TRAINING_COL_NAME_PREFIX = "pred_"
+
+
+def create_preprocessing_pipelines(cfg):
+    """create preprocessing pipeline based on config."""
+    steps = []
+    dtconverter = DateTimeConverter(convert_to=cfg.training.convert_datetimes_to)
+    steps.append("DateTimeConverter", dtconverter)
```

Comment:
```suggestion
    steps.append({"DateTimeConverter", dtconverter})
```

Should be a dict? Or another type?